dianna/utils/maskers.py

```diff
@@ -97,57 +96,185 @@ def _get_mask_value(data: np.array, mask_type: str) -> int:
 
     Args:
         input_size (int): Size of a single sample of input data, for images without the channel axis.
-        n_masks: Number of masks
-        p_keep: Fraction of input data to keep in each mask
-        feature_res (int): Resolution of features in masks.
+        p_keep: ?
+        number_of_masks: Number of masks
+        number_of_features: Number of features per dimension
 
     Returns:
         The generated masks (np.ndarray)
     """
-    cell_size = np.ceil(np.array(input_size) / feature_res)
-    up_size = (feature_res + 1) * cell_size
+    grid = np.random.choice(a=(True, False),
+                            size=(number_of_masks, number_of_features,
+                                  number_of_features),
+                            p=(p_keep, 1 - p_keep)).astype('float32')
+    cell_size = np.ceil(np.array(input_size) / number_of_features)
+    up_size = (number_of_features + 1) * cell_size
+    masks = np.empty((number_of_masks, *input_size), dtype=np.float32)
+    for i in range(masks.shape[0]):
+        y_offset = np.random.randint(0, cell_size[0])
+        x_offset = np.random.randint(0, cell_size[1])
+        # Linear upsampling and cropping
+        masks[i, :, :] = _upscale(grid[i],
+                                  up_size)[y_offset:y_offset + input_size[0],
+                                           x_offset:x_offset + input_size[1]]
+    masks = masks.reshape(-1, *input_size, 1)
+    return masks
 
-    grid = np.random.choice(
-        a=(True, False),
-        size=(n_masks, feature_res, feature_res),
-        p=(p_keep, 1 - p_keep),
-    )
-    grid = grid.astype('float32')
 
-    masks = np.empty((n_masks, *input_size), dtype=np.float32)
+def _generate_interpolated_float_masks_for_timeseries(input_size: int, number_of_masks: int, number_of_features: int) \
+        -> ndarray:
+    """Generates a set of random masks to mask the input data.
 
-    for i in range(n_masks):
-        y = np.random.randint(0, cell_size[0])
-        x = np.random.randint(0, cell_size[1])
-        # Linear upsampling and cropping
-        masks[i, :, :] = _upscale(grid[i], up_size)[y:y + input_size[0],
-                                                    x:x + input_size[1]]
-    masks = masks.reshape(-1, *input_size, 1)
+    Args:
+        input_size (int): Size of a single sample of input time series.
+        number_of_masks: Number of masks
+        number_of_features: Number of features in the time dimension
+
+    Returns:
+        The generated masks (np.ndarray)
+    """
+    grid = np.random.random(size=(number_of_masks, number_of_features,
+                                  1), ).astype('float32')
+
+    masks_shape = (number_of_masks, *input_size)
+
+    if grid.shape == masks_shape:
+        masks = grid
+    else:
+        masks = _project_grids_to_masks(grid, masks_shape)
+    return masks.reshape(-1, *input_size, 1)
+
+
+def _project_grids_to_masks__old(grid: ndarray, masks_shape: tuple,
```

Comment:
Nice to have both the old and new one, for comparison. The old function can be deleted afterward. (action point)

Reply:
deleted